File: radar-commons/src/test/java/org/radarbase/util/Base64Test.java
package org.radarbase.util;

import static org.junit.Assert.*;

import java.util.concurrent.ThreadLocalRandom;
import kotlin.text.Charsets;
import org.junit.Test;
import org.radarbase.util.Base64.Encoder;

public class Base64Test {
    @Test
    public void encoderTest() {
        Encoder encoder = Base64.getEncoder();
        java.util.Base64.Encoder javaEncoder = java.util.Base64.getEncoder();

        ThreadLocalRandom random = ThreadLocalRandom.current();
        for (int i = 0; i < 2_000; i++) {
            byte[] src = new byte[i];
            random.nextBytes(src);
            String actual = encoder.encode(src);
            String expected = new String(javaEncoder.encode(src), Charsets.UTF_8);
            assertEquals(expected, actual);
        }
    }
}

File: radar-commons/src/test/java/org/radarbase/util/Base64Test.kt
package org.radarbase.util

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import java.util.Base64
import java.util.concurrent.ThreadLocalRandom
import kotlin.text.Charsets.UTF_8

class Base64Test {
    @Test
    fun encoderTest() {
        val javaEncoder = Base64.getEncoder()
        val random = ThreadLocalRandom.current()
        var i = 0
        while (i < 2000) {
            val src = ByteArray(i)
            random.nextBytes(src)
            assertEquals(
                Base64Encoder.encode(src),
                String(javaEncoder.encode(src), UTF_8),
            )
            i += 7
        }
    }
}
